badger/badger.go

```diff
@@ -78,21 +85,29 @@ func (b *Datastore) Delete(ctx context.Context, key []byte) error {
 }
 
 func (b *Datastore) Close() error {
+	b.closeLk.Lock()
+	defer b.closeLk.Unlock()
+	b.closed = true
+
 	return b.db.Close()
 }
 
-func (b *Datastore) Iterator(ctx context.Context, iterOpts corekv.IterOptions) corekv.Iterator {
+func (b *Datastore) Iterator(ctx context.Context, iterOpts corekv.IterOptions) (corekv.Iterator, error) {
```

Comment:
nitpick: We have a mix of `b` and `d` receivers for the same struct. `b` seems weird as there is not even a `b` in `Datastore`.  `d` seems a lot more appropriate.

Reply:
The type name used to be `badgerStore` or something similar, the receiver names never got updated when I changed the type name.  I'm not terribly fussed about it given the tiny scope of these functions, and the fact that a lot of reciever and type names are still a little off in this repo (e.g. `bTxn`, slightly further down in this file) - could maybe sort them all out in another PR if/when prioritised.